File: android-mvvm14-viewmodel_livedata/app/src/main/java/com/catalin/yt_mvvm_cities/view/MainActivity.java
package com.catalin.yt_mvvm_cities.view;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.res.ResourcesCompat;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import com.catalin.yt_mvvm_cities.R;
import com.catalin.yt_mvvm_cities.databinding.ActivityMainBinding;
import com.catalin.yt_mvvm_cities.model.City;
import com.catalin.yt_mvvm_cities.viewmodel.CityViewModel;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private CityViewModel model;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        model = new ViewModelProvider(this).get(CityViewModel.class);
    }

    @Override
    protected void onResume() {
        super.onResume();

        model.getCityData().observe(this, new Observer<City>() {
            @Override
            public void onChanged(City city) {
                binding.cityImage.setImageDrawable(
                        ResourcesCompat.getDrawable(getResources() ,city.getImg() , getApplicationContext().getTheme())
                );

                binding.cityNameTV.setText(city.getName());
                binding.cityPopuliationTV.setText(String.valueOf(city.getPopulation()));
            }
        });
    }
}

File: android-mvvm12-viewbinding/app/src/main/java/com/s3d/myviewbinding/MainActivity.kt
package com.s3d.myviewbinding

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.s3d.myviewbinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private var counter = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }

    override fun onStart() {
        super.onStart()
        binding.incrementButton.setOnClickListener {
            counter++
            binding.counter.text = counter.toString()
        }
    }
}